worlds/sm/__init__.py

```diff
@@ -78,12 +78,11 @@ class SMWorld(World):
 
     game: str = "Super Metroid"
     topology_present = True
-    data_version = 1
+    data_version = 2
     option_definitions = sm_options
-    item_names: Set[str] = frozenset(items_lookup_name_to_id)
-    location_names: Set[str] = frozenset(locations_lookup_name_to_id)
-    item_name_to_id = items_lookup_name_to_id
-    location_name_to_id = locations_lookup_name_to_id
+
+    item_name_to_id = dict((value.Name, items_start_id + value.Id) for key, value in ItemManager.Items.items() if value.Id != None)
+    location_name_to_id = dict((key, locations_start_id + value.Id) for key, value in locationsDict.items() if value.Id != None)
```

Comment:
```suggestion
    item_name_to_id = {value.Name: items_start_id + value.Id for key, value in ItemManager.Items.items() if value.Id is not None}
    location_name_to_id = {key: locations_start_id + value.Id for key, value in locationsDict.items() if value.Id is not None}
```

made it a bit more "pythonic"